python/ray/experimental/streaming/benchmarks/micro/batched_queue_benchmark.py

```diff
@@ -0,0 +1,168 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import argparse
+import logging
+import statistics
+import time
+
+import ray
+from ray.experimental.streaming.batched_queue import BatchedQueue
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO)
+
+parser = argparse.ArgumentParser()
+parser.add_argument("--rounds", default=10,
+                    help="the number of experiment rounds")
+parser.add_argument("--num-queues", default=1,
+                    help="the number of queues in the chain")
+parser.add_argument("--queue-size", default=10000,
+                    help="the queue size in number of batches")
+parser.add_argument("--batch-size", default=1000,
+                    help="the batch size in number of elements")
+parser.add_argument("--flush-timeout", default=0.001,
+                    help="the timeout to flush a batch")
+parser.add_argument("--prefetch-depth", default=10,
+                    help="the number of batches to prefetch from plasma")
+parser.add_argument("--background-flush", default=False,
+                    help="whether to flush in the backrgound or not")
+parser.add_argument("--max-throughput", default="inf",
+                    help="maximum read throughput (elements/s)")
+
+@ray.remote
+class Node(object):
+    """An actor that reads from an input queue and writes to an output queue.
+
+    Attributes:
+        id (int): The id of the actor.
+        queue (BatchedQueue): The input queue.
+        out_queue (BatchedQueue): The output queue.
+        max_reads_per_second (int): The max read throughput (default: inf).
+        num_reads (int): Number of elements read.
+        num_writes (int): Number of elements written.
+    """
+    def __init__(self, id, in_queue, out_queue,
+                    max_reads_per_second=float("inf")):
+        self.id = id
+        self.queue = in_queue
+        self.out_queue = out_queue
+        self.max_reads_per_second = max_reads_per_second
+        self.num_reads = 0
+        self.num_writes = 0
+        self.start = time.time()
+
+    def read_write_forever(self):
+        debug_log = "[actor {}] Reads throttled to {} reads/s"
+        log = ""
+        if self.out_queue is not None:
+            self.out_queue.enable_writes()
+            log += "[actor {}] Reads/Writes per second {}"
+        else:   # It's just a reader
```

Comment:
We've been following a convention of using exactly two spaces before the `#`. This may apply in other places as well.

```suggestion
        else:  # It's just a reader
```